src/test/java/com/google/api/codegen/testdata/library_config.baseline

```diff
@@ -1,99 +1,242 @@
+    retry_codes_name: non_idempotent
+    retry_params_name: default
+    request_object_method: false
+    field_name_patterns:
+      name: shelf
   - name: PublishSeries
     flattening:
       groups:
       - parameters:
         - shelf
         - books
         - edition
+    required_fields:
+    - shelf
+    - books
+    - edition
+    retry_codes_name: non_idempotent
+    retry_params_name: default
+    request_object_method: false
+    field_name_patterns: {}
   - name: GetBook
     flattening:
       groups:
       - parameters:
         - name
+    required_fields:
+    - name
+    retry_codes_name: idempotent
+    retry_params_name: default
+    request_object_method: false
+    field_name_patterns:
+      name: book
   - name: ListBooks
     flattening:
       groups:
       - parameters:
         - name
         - filter
+    required_fields:
+    - name
+    - filter
     page_streaming:
       request:
         page_size_field: page_size
         token_field: page_token
       response:
         token_field: next_page_token
-        resource_field: books
+        resources_field: books
+    retry_codes_name: idempotent
+    retry_params_name: default
+    request_object_method: false
+    field_name_patterns:
+      name: shelf
   - name: DeleteBook
     flattening:
       groups:
       - parameters:
         - name
+    required_fields:
+    - name
+    retry_codes_name: idempotent
+    retry_params_name: default
+    request_object_method: false
+    field_name_patterns:
+      name: book
   - name: UpdateBook
+    flattening:
+      groups:
+      - parameters:
+        - name
+        - book
+        - update_mask
+        - physical_mask
+    required_fields:
+    - name
+    - book
+    - update_mask
+    - physical_mask
+    retry_codes_name: idempotent
+    retry_params_name: default
+    request_object_method: false
+    field_name_patterns:
+      name: book
   - name: MoveBook
     flattening:
       groups:
       - parameters:
         - name
         - other_shelf_name
+    required_fields:
+    - name
+    - other_shelf_name
+    retry_codes_name: non_idempotent
+    retry_params_name: default
+    request_object_method: false
+    field_name_patterns:
+      name: book
   - name: ListStrings
     flattening:
       groups:
       - parameters:
         - name
+    required_fields:
+    - name
     page_streaming:
       request:
         page_size_field: page_size
         token_field: page_token
       response:
         token_field: next_page_token
-        resource_field: strings
+        resources_field: strings
+    retry_codes_name: idempotent
+    retry_params_name: default
+    request_object_method: false
+    field_name_patterns: {}
   - name: AddComments
     flattening:
       groups:
       - parameters:
         - name
         - comments
+    required_fields:
+    - name
+    - comments
+    retry_codes_name: non_idempotent
+    retry_params_name: default
+    request_object_method: false
```

Comment:
This is wrong - request_object_method should be false only when there is 0 or 1 parameters, but true otherwise.